### PR TITLE
Attempt to build using Scala 3 + CE3

### PR DESCRIPTION
**Pull request by gvolpe**
Not sure what's the story with compiler plugins in Scala 3 but we might need to get rid of `context-applied`.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,7 +1,18 @@
 import Dependencies._
 import Settings._
 
-scalaVersion in ThisBuild := "2.13.2"
+ThisBuild / crossScalaVersions := Seq("2.13.2", "3.0.0-M2")
```
Comment (psisoyev) [inline build.sbt:4]: There is already M3 available :)

Comment (psisoyev) [reply build.sbt:4]: Also, can we move to 2.13.4? Or it would require additional work?

Comment (gvolpe) [reply build.sbt:4]: Yeah but there's no Fs2 dependency for `3.0.0-M3` yet. The current branch does not compile with Scala 2.13.4 and I don't want do dig down the rabbit hole right now...

Comment (psisoyev) [reply build.sbt:4]: That's right, it's not worth it right now and should be done separately

File: build.sbt [hunk 2]
```diff
@@ -34,21 +44,20 @@ lazy val `neutron-circe` = (project in file("circe"))
...
-          Libraries.java8Compat,
-          Libraries.newtype,
-          Libraries.cats            % Test,
-          Libraries.catsEffect      % Test,
-          Libraries.munitCore       % Test,
-          Libraries.munitScalacheck % Test,
-          Libraries.cats            % Test
-        )
-  )
+//lazy val `neutron-function` = (project in file("function"))
```
Comment (psisoyev) [inline build.sbt:47]: Do you have any issues with this module?

**Pull request closed without merging**